agent/sighandlers/termination_handler.go

```diff
@@ -87,8 +88,9 @@ func FinalSave(saver statemanager.Saver, taskEngine engine.TaskEngine) error {
 	})
 	go func() {
 		seelog.Debug("Saving state before shutting down")
-		stateSaved <- saver.ForceSave()
+		saveStateAll(state, dataClient)
 		saveTimer.Stop()
+		stateSaved <- nil
```

Comment:
just curious, it seems `stateSaved` is used for error reporting, after this change, the return code (`stateSaved`) will always be `nil`, is that expected? Are we handling error from `saveStateAll` somewhere else?

Reply:
stateSave channel will still receive error when saveTimer times out. 
If there is any error in saveStateAll we are just going to log them(similar to https://github.com/aws/amazon-ecs-agent/blob/boltdb/agent/engine/data.go#L27-L30)